tests/bool/bad/test_xor_with_int.catala_en

```diff
@@ -29,5 +29,24 @@ $ catala Typecheck
 │ 8 │   definition test_var equals 10 xor 20
 │   │                                 ‾‾‾
 └─ 'xor' should be a boolean operator
+┌─[ERROR]─
+│
+│  Error during typechecking, incompatible types:
+│  ─➤ integer
+│  ─➤ bool
+│
+│ This expression has type integer:
+├─➤ tests/bool/bad/test_xor_with_int.catala_en:8.37-8.39:
+│   │
+│ 8 │   definition test_var equals 10 xor 20
+│   │                                     ‾‾
+├─ 'xor' should be a boolean operator
+│
+│ Expected type bool coming from expression:
+├─➤ tests/bool/bad/test_xor_with_int.catala_en:8.33-8.36:
+│   │
+│ 8 │   definition test_var equals 10 xor 20
+│   │                                 ‾‾‾
+└─ 'xor' should be a boolean operator
 #return code 123#
 ```
```

Comment:
Hm, this is a bit redundant... Maybe we could filter type errors that are "too close" to each other?
Or something much simpler could be to just give up typing of the current definition, and keep typing the rest of the program... since all types need to be declared, it wouldn't propagate errors.

Just thinking about the different options, it's good to experiment here.

Reply:
Note that in this case, this might be tweaked by changing the order in which typing is done (e.g. typing the arguments first, then the operator, rather than the opposite ; ofc it would be worse on `true xor 42`)